### PR TITLE
libjxl: build against system lcms2

### PR DESCRIPTION
**Pull request by yu-re-ka**
prevent accidentally building against vendored libs. Fixes build of pkgsMusl.libjxl.

Change-Id: I8f1cba112991b47f0951acc2a2471757a9c440bb

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by emilazy** [comment]
I think it’s probably for the best to, as we’ve had some uncomfortably large channel bumps recently and Hydra has been on fire a little bit with the release.

File: pkgs/by-name/li/libjxl/package.nix [hunk 1]
```diff
@@ -121,6 +124,11 @@ stdenv.mkDerivation rec {
   # the second substitution fix regex for a2x script
   # https://github.com/libjxl/libjxl/pull/3842
   postPatch = ''
+    # Make sure we do not accidentally build against some of the vendored dependencies
+    # If it asks you to "run deps.sh to fetch the build dependencies", then you are probably missing a JPEGXL_FORCE_SYSTEM_* flag
+    shopt -s extglob
+    rm -rf third_party/!(sjpeg)/
```
Comment (emilazy) [inline pkgs/by-name/li/libjxl/package.nix:130]: I like to use subshells when setting shell options to not risk disrupting any glue code / later phases and hooks / etc.
```suggestion
    (
      shopt -s extglob
      rm -rf third_party/!(sjpeg)/
    )
```

**Review by emilazy — APPROVED** [review]
Builds on `aarch64-linux` and I’ll trust you on Musl; thanks! Will merge tomorrow if I remember to; one *non‐blocking* nit (sorry for not looking over the patch more thoroughly before suggesting the retarget).

**Comment by trofi** [comment]
The change broke `patchPhase`:
- https://github.com/NixOS/nixpkgs/pull/359952

**Pull request merged**